ingester/src/data.rs

```diff
@@ -79,48 +93,197 @@ pub struct IngesterData {
+                    partition_info.partition.sequencer_id
+                )
+            }); //{
+        let namespace = sequencer_data
+            .namespace(&partition_info.namespace_name)
+            .unwrap_or_else(|| {
+                panic!(
+                    "namespace {} not in sequencer {} state",
+                    partition_info.namespace_name, partition_info.partition.sequencer_id
+                )
+            });
+        let table_data = namespace
+            .table_data(&partition_info.table_name)
+            .unwrap_or_else(|| {
+                panic!(
+                    "table {} for namespace {} not in sequencer {} state",
+                    partition_info.table_name,
+                    partition_info.namespace_name,
+                    partition_info.partition.sequencer_id
+                )
+            });
+        let partition_data = table_data
+            .partition_data(&partition_info.partition.partition_key)
+            .unwrap_or_else(|| {
+                panic!(
+                    "partition {} not in table {} for namespace {} in sequencer {} state",
+                    partition_info.partition.partition_key,
+                    partition_info.table_name,
+                    partition_info.namespace_name,
+                    partition_info.partition.sequencer_id
+                )
+            });
+
+        // snapshot and make arc clones of the data.
+        let persisting_batch = partition_data.snapshot_to_persisting_batch(
+            partition_info.partition.sequencer_id,
+            partition_info.partition.table_id,
+            partition_info.partition.id,
+            &partition_info.table_name,
+        );
+
+        // do the CPU intensive work of compaction, de-duplication and sorting
+        let (record_batches, iox_meta) = match compact_persisting_batch(
+            Arc::new(SystemProvider::new()),
+            &self.exec,
+            namespace.namespace_id.get(),
+            &partition_info.namespace_name,
+            &partition_info.table_name,
+            &partition_info.partition.partition_key,
+            Arc::clone(&persisting_batch),
+        )
+        .await
+        {
+            Err(e) => {
+                // this should never error out. if it does, we need to crash hard so
+                // someone can take a look.
+                panic!("unable to compact persisting batch with error: {:?}", e);
+            }
+            Ok(Some(r)) => r,
+            Ok(None) => {
+                warn!("persist called with no data");
+                return;
+            }
+        };
+
+        // save the compacted data to a parquet file in object storage
+        loop {
+            match persist(&iox_meta, record_batches.to_vec(), &self.object_store).await {
+                Ok(_) => break,
+                Err(e) => {
+                    error!(%e, "error persisting parquet file to object store");
```

Comment:
I assume this behaviour covers things like networking issues or other transient issues in the CSP? I don't think `error` is warranted here.

```suggestion
                    warn!(%e, "persisting to object store failed: retrying.");
```